enforcer/src/main/java/org/wso2/choreo/connect/enforcer/analytics/AnalyticsEventPublisher.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.choreo.connect.enforcer.analytics;
+
+import io.envoyproxy.envoy.service.accesslog.v3.StreamAccessLogsMessage;
+
+import java.util.Map;
+
+/**
+ * AnalyticsEventPublisher interface should be used to write a custom analytics publisher.
+ */
+public interface AnalyticsEventPublisher {
```

Comment:
Try to move it to carbon.apimgt.common package

Reply:
Since this is only used in choreo connect, decided against moving it to the carbon apimgt common package.